src/Microsoft.PowerShell.ConsoleGuiTools/ConsoleGui.cs

```diff
@@ -120,9 +130,21 @@ private GridViewDataSource LoadData()
 
         private void ApplyFilter()
         {
-            List<GridViewRow> itemList = GridViewHelpers.FilterData(_itemSource.GridViewRowList, _applicationData.Filter ?? string.Empty);
-            // Set the ListView to show only the subset defined by the filter
-            _listView.Source = new GridViewDataSource(itemList);
+            // The ListView is always filled with a (filtered) copy of _inputSource.
+            // We listen for `MarkChanged` events on this filtered list and apply those changes up to _inputSource.
+
+            if (_listViewSource != null) {
+                _listViewSource.MarkChanged -= ListViewSource_MarkChanged;
+                _listViewSource = null;
```

Comment:
Just noting that I think we're relying on the garbage collector to dispose of the previously created `_listViewSource` here. I _think_ that's fine? If `_listViewSource` is disposable, we should do that. I'm not sure if it is though...will find out soon I guess.

Reply:
Ok we're good! It is not `Disposable`, it's an implementation of `IListDataSource` (which is a simple interface with just primitives and methods) and it holds a `List<GridViewRow>` which is managed only and thus not disposable.